packages/preferences/src/browser/preferences-tree-widget.ts

```diff
@@ -0,0 +1,276 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject } from "inversify";
+import { Message } from "@phosphor/messaging";
+import { PreferencesMenuFactory } from "./preferences-menu-factory";
+import { PreferencesDecorator } from "./preferences-decorator";
+import {
+    ContextMenuRenderer,
+    ExpandableTreeNode,
+    PreferenceSchemaProvider,
+    SelectableTreeNode,
+    PreferenceProperty,
+    PreferenceScope,
+    PreferenceService,
+    TreeModel,
+    TreeNode,
+    TreeProps,
+    TreeWidget,
+    WidgetManager,
+    Saveable
+} from "@theia/core/lib/browser";
+import {
+    USER_PREFERENCES_TREE_WIDGET_ID,
+    WORKSPACE_PREFERENCES_TREE_WIDGET_ID
+} from "./preferences-contribution";
+import { UserPreferenceProvider } from "./user-preference-provider";
+import { WorkspacePreferenceProvider } from "./workspace-preference-provider";
+import { EditorWidget } from "@theia/editor/lib/browser";
+import { DisposableCollection, Emitter, Event, MaybePromise } from '@theia/core';
+import { Widget } from "@phosphor/widgets";
+import { EditorBasedSplitPanel } from "@theia/editor/lib/browser/editor-widget";
+
+export abstract class PreferencesContainer extends EditorBasedSplitPanel implements Saveable {
```

Comment:
@AlexTugarev 
While reworking the user and workspace preferences widgets to single one I've faced with a problem: What to do with the dirty-state indicator? Since the widgets are placed in the common panel they don't have the close icon.
![screenshot-localhost-3000-2018 07 25-14-44-55](https://user-images.githubusercontent.com/7668752/43199042-5990b130-9019-11e8-9643-fea10d2536fc.png)

Reply:
@vinokurig I meant to have a tab folder for the code editor part only.

Reply:
try to save on change between both parts. e.g. having changes in user preferences and switching to workspace preferences would save user preferences.

Reply:
the tree part should stay stable, I think. therefore, please try with the tab folder above the code editors.

Reply:
So dirty indicator is missed, right?